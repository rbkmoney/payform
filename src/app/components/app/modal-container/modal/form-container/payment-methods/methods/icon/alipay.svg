<svg viewBox="80 50 1150 1100" xmlns="http://www.w3.org/2000/svg">
    <path
        fill="#685BFF"
        clip-rule="evenodd"
        fill-rule="evenodd"
        d="M995.853568 770.303464c-48.84605-16.282017-114.478984-41.273019-187.432515-67.65241 43.923579-76.108961 78.885739-162.820165 101.857266-256.978183H669.582152v-86.584987h294.843338v-48.341181H669.582152V166.606681h-120.411192c-21.078269 0-21.078269 20.825835-21.078269 20.825835v123.440404H229.967706v48.341181h298.251202v86.584987H281.969185v48.341181h477.605818a842.095797 842.095797 0 0 1-68.788364 167.742635c-154.9947-50.991742-320.339209-92.390977-424.21595-66.895107-66.390238 16.408234-109.177863 45.564403-134.421299 76.108961-115.362505 140.101072-32.69025 352.903242 211.035129 352.903242C487.198324 1023.873783 626.037224 943.599655 733.574264 811.576482 893.996302 888.568963 1211.68495 1020.718353 1211.68495 1020.718353V832.2761s-39.88463-3.15543-215.831382-61.972636zM320.465426 942.337483c-189.956859 0-246.123505-149.314927-152.217922-231.103661 31.301861-27.641563 88.604462-41.146801 119.14902-44.176014 112.838161-11.107112 217.345988 31.80673 340.660175 91.886109-86.837421 112.964378-197.277456 183.393566-307.591273 183.393566z"
    ></path>
</svg>
