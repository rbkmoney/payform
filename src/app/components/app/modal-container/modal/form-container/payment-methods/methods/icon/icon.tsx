import * as React from 'react';

import styled from 'checkout/styled-components';
import ApplePay from './apple-pay.svg';
import BankCard from './bank-card.svg';
import GooglePay from './google-pay.svg';
import SamsungPay from './samsung-pay.svg';
import Terminals from './terminals.svg';
import Wallets from './wallets.svg';
import Mobile from './mobile.svg';

const IconClasses = {
    'apple-pay': ApplePay,
    'google-pay': GooglePay,
    'samsung-pay': SamsungPay,
    'bank-card': BankCard,
    terminals: Terminals,
    wallets: Wallets,
    mobile: Mobile
};

type name = keyof typeof IconClasses;

export const Icon = styled<React.FC<{ name: name; className?: string }>>((props) => {
    const IconClass = IconClasses[props.name];
    return (
        <div className={props.className}>
            <IconClass />
        </div>
    );
})`
    height: 40px;
    width: 40px;
    margin-right: 15px;

    path {
        fill: ${({ theme }) => theme.color.secondary[0.9]};
    }
`;
