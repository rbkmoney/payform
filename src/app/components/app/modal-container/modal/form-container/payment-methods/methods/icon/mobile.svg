<svg width="40" height="40" fill="none" xmlns="http://www.w3.org/2000/svg"><path clip-rule="evenodd" d="M26.204 34H13.797A2.796 2.796 0 0111 31.204V7.796A2.796 2.796 0 0113.797 5h12.407A2.796 2.796 0 0129 7.796v23.408A2.796 2.796 0 0126.204 34z" stroke="#685BFF" stroke-width="2"/><mask id="a" maskUnits="userSpaceOnUse" x="0" y="0" width="40" height="40"><path fill-rule="evenodd" clip-rule="evenodd" d="M0 40h40V0H0v40z" fill="#fff"/></mask><g mask="url(#a)" fill-rule="evenodd" clip-rule="evenodd" fill="#685BFF"><path d="M14 26h12V8H14v18zm1-1h10V9H15v16zM18 29h4v-1h-4v1z"/></g></svg>