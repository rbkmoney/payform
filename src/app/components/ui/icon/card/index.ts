export { default as Mastercard } from './mastercard.svg';
export { default as Visa } from './visa.svg';
export { default as Mir } from './mir.svg';
export { default as Maestro } from './maestro.svg';
export { default as AmericanExpress } from './american-express.svg';
export { default as DinersClub } from './diners-club.svg';
export { default as Discover } from './discover.svg';
export { default as Elo } from './elo.svg';
export { default as Hipercard } from './hipercard.svg';
export { default as Unionpay } from './unionpay.svg';
export { default as Jcb } from './jcb.svg';
