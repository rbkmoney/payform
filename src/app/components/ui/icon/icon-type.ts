export enum IconType {
    calendar = 'calendar',
    checkmark = 'checkmark',
    redCross = 'red-cross',
    chevronLeft = 'chevron-left',
    amount = 'amount',
    cross = 'cross',
    card = 'card',
    letter = 'letter',
    lock = 'lock',
    phone = 'phone',
    user = 'user',
    visa = 'visa',
    mastercard = 'mastercard',
    eurosetLogo = 'euroset-logo'
}
