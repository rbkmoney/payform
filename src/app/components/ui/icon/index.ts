export { default as Amount } from './amount.svg';
export { default as BoldCross } from './bold-cross.svg';
export { default as Calendar } from './calendar.svg';
export { default as Card } from './card.svg';
export { default as Checkmark } from './checkmark.svg';
export { default as ChevronLeft } from './chevron-left.svg';
export { default as Cross } from './cross.svg';
export { default as EurosetLogo } from './euroset-logo.svg';
export { default as Letter } from './letter.svg';
export { default as Lock } from './lock.svg';
export { default as Mastercard } from './mastercard.svg';
export { default as Phone } from './phone.svg';
export { default as User } from './user.svg';
export { default as Visa } from './visa.svg';
export { default as Mir } from './mir.svg';
