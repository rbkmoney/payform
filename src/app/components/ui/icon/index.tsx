export * from './icon';
export * from './icon-type';
