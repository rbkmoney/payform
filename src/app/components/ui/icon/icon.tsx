import * as React from 'react';
import { IconType } from './icon-type';

interface IconProps {
    icon: IconType;
    className?: string;
}

export const Icon: React.SFC<IconProps> = (props) => (
    <svg className={props.className}>
        <use xlinkHref={`assets/icons/${props.icon}.svg#icon`}/>
    </svg>
);
