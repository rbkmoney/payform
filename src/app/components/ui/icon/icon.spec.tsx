import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Icon } from './icon';
import { IconType } from './icon-type';

it('renders correctly', () => {
    const tree = renderer.create(<Icon className={'test-class'} icon={IconType.amount} />).toJSON();
    expect(tree).toMatchSnapshot();
});
