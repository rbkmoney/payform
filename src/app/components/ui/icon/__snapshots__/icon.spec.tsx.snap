// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<svg
  className="test-class"
>
  <use
    xlinkHref="assets/icons/amount.svg#icon"
  />
</svg>
`;
